02-microarray/ortholog-mapping_microarray_01_ensembl.Rmd

```diff
@@ -0,0 +1,413 @@
+  dplyr::group_by(human_symbol) %>%
+  dplyr::count() %>%
+  # Sort by highest `n` which would be the human gene symbol with the most
+  # mapped zebrafish Ensembl IDs
+  dplyr::arrange(desc(n))
+```
+
+Looks like we have mapped symbols!
+
+Now, let's get an idea of how many zebrafish Ensembl IDs we have that were not mapped to human gene symbols.
+
+```{r}
+sum(is.na(human_zebrafish_mapped_df$human_symbol))
+```
+
+We have 463 NAs, which means we have 463 zebrafish Ensembl IDs that were not mapped to human gene symbols.
+This is okay because we do not expect everything to map across species.
+
+## Take a look at some multi-mappings
+
+If a zebrafish Ensembl gene ID maps to multiple human symbols, the associated values will get duplicated.
+Let's look at the `ENSDARG00000069142` example below.
+
+```{r}
+human_zebrafish_mapped_df %>%
+  dplyr::filter(Gene == "ENSDARG00000069142")
+```
+
+On the other hand, if you were to look at the original data associated to the zebrafish Ensembl IDs, when a human gene symbol maps to multiple zebrafish Ensembl IDs, the values will not get duplicated, but you will have multiple rows associated with that human symbol.
+Let's look at the `MATR3` example below.
+
+```{r}
+human_zebrafish_mapped_df %>%
+  dplyr::filter(human_symbol == "MATR3")
+```
+
+## Collapse zebrafish genes mapping to multiple human genes
+
+Remember that if a zebrafish Ensembl gene ID maps to multiple human symbols, the values get duplicated.
+We can collapse the multi-mapped values into a list for each Ensembl ID as to not have duplicate values in our data frame.
+
+In the next chunk, we show how we can collapse all the human gene symbols into one column where we store them all for each unique zebrafish Ensembl ID.
+
+```{r}
+collapsed_human_symbol_df <- human_zebrafish_mapped_df %>%
+  # Group by zebrafish Ensembl IDs
+  dplyr::group_by(Gene) %>%
+  # Collapse the mapped values in `human_zebrafish_mapped_df` into one column named
+  # `all_human_symbols` -- note that we will lose the `support` column in this summarizing step
+  dplyr::summarize(all_human_symbols = paste(human_symbol, collapse = ";"))
+
+head(collapsed_human_symbol_df)
+```
+
+### Write results to file
+
+Now let's write our list of human gene symbols for each unique zebrafish Ensembl ID results to file.
+
+```{r}
+readr::write_tsv(
+  collapsed_human_symbol_df,
+  file.path(
+    results_dir,
+    # Replace with a relevant output file name
+    "GSE71270_zebrafish_ensembl_to_collapsed_human_gene_symbol.tsv"
+  )
+)
+```
+
+## Collapse human genes mapping to multiple zebrafish genes
+
+Since multiple zebrafish Ensembl gene IDs map to the same human symbol, we can also collapse all Ensembl IDs that each unique human gene symbol mapped to.
+
+A known issue in ortholog mapping is how to handle this data because combining values across zebrafish transcripts using an average may result in the loss of a lot of data and will likely not be representative of the zebrafish biology.
```

Comment:
I feel like this is still a bit unclear.. What are we trying to say here? What is the goal?

Are we are trying to find the "most similar" zebrafish gene to a given human gene? If so, looking at the level of support across databases might be one way to handle this, but it is not the only one. I think we should state very clearly here that this is not the only, or even recommended,  way to handle these multimappings.

I have written a bit below to try to get across my thoughts on the matter. 

```suggestion
Since multiple zebrafish Ensembl gene IDs map to the same human symbol, we may want to identify which one of these mappings represents the "true" ortholog, i.e. which zebrafish gene is most similar to the human gene we are interested in. 
*This is not at all straightforward!* (see, for just one example, https://doi.org/10.1093/bioinformatics/btaa468)
Gene duplications along the zebrafish lineage may result in complicated relationships among genes, especially with regard to divisions of function.

Simply combining values across zebrafish transcripts using an average may result in the loss of a lot of data and will likely not be representative of the zebrafish biology.
One thing we might do to make the problem somewhat simpler is to reduce the number of multimapped genes by requiring a certain level of support for each mapping from across the various databases included in `HCOP`.
This will not fully solve the problem (and may not even be desirable in some cases), but we present it here as an example of an approach one might take.
```